salt/modules/openbsd_sysctl.py

```diff
@@ -126,7 +130,7 @@ def persist(name, value, config='/etc/sysctl.conf'):
     if not edited:
         nlines.append('{0}={1}\n'.format(name, value))
     with salt.utils.files.fopen(config, 'w+') as ofile:
-        ofile.writelines(nlines)
+        ofile.writelines(salt.utils.data.decode(nlines))
```

Comment:
This should be `encode`, not `decode`, and the filehandle should be opened with a mode of `wb`.